src/Data/Algorand/Transaction.hs

```diff
@@ -118,29 +113,60 @@ data TransactionType
   deriving (Eq, Generic, Show)
 
 -- | Constants for @OnComplete@.
---
-type OnComplete = Text
-
-onCompleteNoOp :: OnComplete
-onCompleteNoOp = "call"
-
-onCompleteCreate :: OnComplete
-onCompleteCreate = "create"
-
-onCompleteOptIn :: OnComplete
-onCompleteOptIn = "optin"
-
-onCompleteCloseOut :: OnComplete
-onCompleteCloseOut = "closeout"
-
-onCompleteClearState :: OnComplete
-onCompleteClearState = "clear"
-
-onCompleteUpdateApplication :: OnComplete
-onCompleteUpdateApplication = "update"
-
-onCompleteDeleteApplication :: OnComplete
-onCompleteDeleteApplication = "delete"
+data OnComplete
+  = OnCompleteNoOp
+  | OnCompleteOptIn
+  | OnCompleteCloseOut
+  | OnCompleteClearState
+  | OnCompleteUpdateApplication
+  | OnCompleteDeleteApplication
+  deriving (Eq, Show)
+
+instance Default OnComplete where
+  def = OnCompleteNoOp
+instance CanonicalZero OnComplete
+instance NonZeroValue OnComplete
+
+instance Enum OnComplete where
```

Comment:
Can't it be derived instead?

Reply:
Yeah, seems like it can be derived with no problem, I update this now.